substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/linux/LinuxThreadCpuTimeSupport.java

```diff
@@ -34,22 +35,49 @@
 import com.oracle.svm.core.posix.headers.Time.timespec;
 import com.oracle.svm.core.posix.headers.linux.LinuxPthread;
 import com.oracle.svm.core.posix.headers.linux.LinuxTime;
+import com.oracle.svm.core.posix.headers.Unistd;
 import com.oracle.svm.core.thread.ThreadCpuTimeSupport;
+import com.oracle.svm.core.thread.VMThreads;
 import com.oracle.svm.core.thread.VMThreads.OSThreadHandle;
 import com.oracle.svm.core.util.TimeUtils;
+import com.oracle.svm.core.threadlocal.FastThreadLocalInt;
+import com.oracle.svm.core.threadlocal.FastThreadLocalFactory;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.BufferedReader;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 @AutomaticallyRegisteredImageSingleton(ThreadCpuTimeSupport.class)
 final class LinuxThreadCpuTimeSupport implements ThreadCpuTimeSupport {
 
+    private static final FastThreadLocalInt kernelThreadId = FastThreadLocalFactory.createInt("LinuxThreadCpuTimeSupport.kernelThreadId");
```

Comment:
I don't think that we need a new thread local. It should be possible to change the existing code so that `VMThreads.OSThreadIdTL` contains the thread id on Linux and macOS (instead of `pthread_t`, which isn't really correct and redundant, see `OSThreadHandleTL`).

Reply:
The code is updated to return `gettid()` from `PosixVMThreads.getCurrentOSThreadId()` method on Linux platform. The thread local is removed from `LinuxThreadCpuTimeSupport` and its `getThreadCpuTime(isolateThread, includeSystemTime)` method now uses OSThreadId to get the thread user time.